File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        n=len(strs)
        new={}
        for i in range(n):
            stn="".join(sorted(strs[i]))
            if stn not in new:
                new[stn]=[]
            new[stn].append(strs[i])
        return list(new.values())

File: 0049-group-anagrams/0049-group-anagrams.java
import java.util.*;
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> dicti = new HashMap<>();
        for(int i=0;i<strs.length;i++){
            String temp=strs[i];
            char[] charArray = temp.toCharArray();
            Arrays.sort(charArray);
            String sortedStr = new String(charArray);
            
            if (dicti.containsKey(sortedStr)){
                dicti.get(sortedStr).add(strs[i]);
            }
            else{
                dicti.put(sortedStr, new ArrayList<>());
                dicti.get(sortedStr).add(strs[i]);
            }
        }
        Collection<List<String>> values = dicti.values();
        List<List<String>> ans = new ArrayList<>();
        for (List<String> value : values) {
            ans.add(value);
        }
        return ans;
    }
}